app/controllers/foods_controller.rb

```diff
@@ -1,14 +1,38 @@
 class FoodsController < ApplicationController
   before_action :authenticate_user!
-  def index; end
+  def index
+    @user = current_user
+    @foods = @user.foods
+  end
 
   def show; end
 
-  def new; end
+  def new
+    @food = Food.new
+  end
 
-  def create; end
+  def destroy
+    @food = current_user.foods.find(params[:id])
+    if @food.destroy
+      redirect_to foods_path, notice: 'Food was successfully destroyed.'
+    else
+      redirect_to foods_path, alert: 'Failed to destroy food.'
+    end
+  end
```

Comment:
[OPTIONAL]
When I try to delete a food relate with a recipe it causes an error due to the foreign key

![image](https://github.com/PabloBona/recipe-app/assets/119966922/2cc12e97-30cc-4633-9f25-56e4bb653374)

Kindly fix this using and alert that mention you can't delete that food, you can try something like this

```ruby
begin
      @food.destroy
      redirect_to foods_path
    rescue ActiveRecord::InvalidForeignKey
      flash[:error] = "Can't delete food due to existing relationships"
      redirect_to foods_path
    end
  end
```

Reply:
![image](https://github.com/PabloBona/recipe-app/assets/110569486/35885ea5-d4a6-4e74-94f9-8d99cd04e9a4)
Fixed!!
Thanks @AndreaM2429 
![image](https://github.com/PabloBona/recipe-app/assets/110569486/65e46f20-33c8-4228-9a8c-10bbb6c11b81)